File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String lowerS = s.toLowerCase();
        int i = 0;
        int j = lowerS.length() -1;

        while (i<j){
            char left = lowerS.charAt(i);
            char right = lowerS.charAt(j);

            if (!(left>='a' && left<='z')){
                if(!(left>= '0' && left<='9')){
                    i++;
                    continue;
                }
            }
            if (!(right>='a' && right<='z')){
                if(!(right>= '0' && right<='9')){
                    j--;
                    continue;
                }
            }

            if (left!=right){
                return false;
            } 
            
            i++;
            j--;
            
        }
        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        string t;
        bool rushil = true;
        for(int i =0; i< s.length(); i++){
            if(!(s[i]>='A' && s[i]<='Z')){
                if(!(s[i]>='a' && s[i]<='z')){
                    if(!(s[i] >= '0' && s[i] <= '9' )){
                        continue;   
                    }
                }
            }
            t += tolower(s[i]);
        }
        if(t.length() == 1) return true;
        int i = 0;
        int j = t.length()-1;
        while (j>i)
        {
            if(t[i] != t[j])
            {
                rushil = false;
                break;
            }
            i++;
            j--;
        }
        if(rushil) return true;
        else return false;
    }
};